aten/src/ATen/SparseCsrTensorUtils.h

```diff
@@ -103,5 +109,19 @@ inline int plainDimension(Layout layout, IntArrayRef size) {
   return size.size() - (isCompressedRow(layout) ? 1 : 2);
 }
 
+template <typename Func, typename... Args>
+Tensor map_sparse_csr_values(const Tensor& t, Func f, Args... args) {
+  TORCH_CHECK(t.is_sparse_csr(), "Expected input to be a SparseCsr Tensor.");
+  auto new_values = f(t.values(), args...);
```

Comment:
I think we have to apply the device-changing-part of the map to `crow_indices` and `col_indices` as well because we currently require that all member tensors use the same device but the following example
```
a = torch.tensor([[0, 1, 2], [2, 0, 3]], device='cuda').to_sparse_csr()
b = a.to(device='cpu')
```
will result a tensor that silently breaks this device invariance: `b.crow_indices()` and `b.col_indices()` use cuda device but `b.values()` uses CPU.

As a solution, correct the indices device if the values device is changed (pseudo-code follows):
```
new_crow_indices = t.crow_indices()
new_col_indices = t.col_indices()
if new_values.device != t.values().device:
    new_crow_indices = new_crow_indices.to(device=new_values.device)
    new_col_indices = new_col_indices.to(device=new_values.device)
```

Reply:
Could you also add a test that hits this invariance breakage?